packages/webcomponents/docs/helpers/ComponentExample.ts

```diff
@@ -0,0 +1,417 @@
+        Object.entries(componentProps).forEach(([key, value]) => {
+          if (typeof value === 'boolean') {
+            if (value) {
+              element.setAttribute(key, '');
+            }
+          } else {
+            element.setAttribute(key, String(value));
+          }
+        });
+      } catch (e) {
+        console.error('Failed to parse component-props:', e);
+      }
+    }
+
+    // Wrap the element in a div with the class component-example-children
+    const wrapper = document.createElement('div');
+    wrapper.className = 'component-example-children';
+    wrapper.appendChild(element);
+    this.container.appendChild(wrapper);
+  }
+
+  private showError(error: string) {
+    // Guard against SSR
+    if (!isBrowser || !document) {
+      return;
+    }
+
+    if (!this.container) {
+      this.container = document.createElement('div');
+      this.appendChild(this.container);
+    }
+
+    this.container.innerHTML = `
+      <div style="padding: 1rem; border: 1px solid #ff6b6b; border-radius: 4px; background-color: #ffe0e0;">
+        <strong>Error loading component:</strong> ${error}
+      </div>
+    `;
+  }
+
+  private showLoading() {
+    // Guard against SSR
+    if (!isBrowser || !document) {
+      return;
+    }
+
+    const componentName = this.getAttribute('component-name') || 'component';
+    if (!this.container) {
+      this.container = document.createElement('div');
+      this.appendChild(this.container);
+    }
+
+    this.container.innerHTML = `
+      <div style="padding: 1rem; text-align: center; color: #666;">
+        Loading ${componentName}...
+      </div>
+    `;
+  }
+
+  private cleanup() {
+    // Guard against SSR
+    if (!isBrowser) {
+      return;
+    }
+
+    if (this.checkInterval) {
+      clearInterval(this.checkInterval);
+      this.checkInterval = null;
+    }
+    // Restore original fetch (only if no other instances are using it)
+    // Note: This is a simplified cleanup - in a real scenario, you might want
+    // to track how many instances are using fetch mocking
+    if (this.originalFetch && typeof window !== 'undefined' && window.fetch) {
+      window.fetch = this.originalFetch;
+      this.originalFetch = null;
+    }
+  }
+}
+
+// Register the custom element (only in browser with customElements support)
+export function registerComponentExample() {
+  // Guard against SSR - only register if customElements exists
+  if (!hasCustomElements || !customElements) {
+    return;
+  }
+
+  if (!customElements.get('component-example')) {
+    customElements.define('component-example', ComponentExampleElement);
+  }
+}
+
+// Auto-register if in browser environment with customElements support
+// Only call if customElements exists (SSR-safe)
+if (hasCustomElements) {
+  registerComponentExample();
+}
+
+// Helper function to get React dynamically (available in MDX/Storybook context)
+export function getReact() {
```

Comment:
This is still trying to use React by finding it in the global context or on the `window`